core/lib/zksync_core/src/basic_witness_input_producer/mod.rs

```diff
@@ -0,0 +1,185 @@
+use std::sync::Arc;
+use std::time::Instant;
+
+use zksync_dal::ConnectionPool;
+use zksync_object_store::{ObjectStore, ObjectStoreFactory};
+use zksync_queued_job_processor::JobProcessor;
+use zksync_types::witness_block_state::WitnessBlockState;
+use zksync_types::{L1BatchNumber, L2ChainId};
+
+use async_trait::async_trait;
+use tokio::runtime::Handle;
+use tokio::task::JoinHandle;
+
+mod vm_interactions;
+
+use vm_interactions::{create_vm, execute_tx};
+use zksync_types::block::MiniblockExecutionMode;
+
+/// Component that extracts all data (from DB) necessary to run a Basic Witness Generator.
+/// This component will upload Witness Inputs to the object store.
+/// This allows Witness Generator workflow (that needs only Basic Witness Generator Inputs)
+/// to be run only using the object store information, having no other external dependency.
+#[derive(Debug)]
+pub struct BasicWitnessInputProducer {
+    connection_pool: ConnectionPool,
+    validation_computational_gas_limit: u32,
```

Comment:
I think we shouldn't keep `validation_computational_gas_limit` as a field given it's always equal to `u32::MAX`. Also, please add a comment everywhere `u32::MAX` is used